### PR TITLE
Fix BIO_sendmmsg/BIO_recvmmsg issues on FreeBSD

### LINKED ISSUE #19156: Test failures on FreeBSD
#19130 fixed a recent compile failure on FreeBSD.

While the compilation failure seems to have been resolved, we still seem to be getting test failures on that platform. E.g. see:

https://ci.buildbot.openssl.org/#/builders/11/builds/9

````
        # INFO:  @ test/bio_dgram_test.c:121
        # # Testing with AF_INET, local=1
        # 
        # ERROR: (bool) 'do_sendmmsg(b1, tx_msg, 2, 0, &num_processed) == true' failed @ test/bio_dgram_test.c:272
        # false
        # 0020E10008000000:error:80000027:system library:dgram_sendmmsg:Destination address required:crypto/bio/bss_dgram.c:1314:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_sendmmsg:local address not available:crypto/bio/bss_dgram.c:1297:
        # 0020E10008000000:error:80000016:system library:dgram_sendmmsg:Invalid argument:crypto/bio/bss_dgram.c:1314:
        # OPENSSL_TEST_RAND_ORDER=1662544797
        not ok 3 - iteration 3
# ------------------------------------------------------------------------------
        # INFO:  @ test/bio_dgram_test.c:127
        # # Testing with AF_INET6, local=1
        # 
        # ERROR: (bool) 'do_recvmmsg(b2, rx_msg, 2, 0, &num_processed) == true' failed @ test/bio_dgram_test.c:323
        # false
        # 0020E10008000000:error:80000027:system library:dgram_sendmmsg:Destination address required:crypto/bio/bss_dgram.c:1314:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_sendmmsg:local address not available:crypto/bio/bss_dgram.c:1297:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_recvmmsg:local address not available:crypto/bio/bss_dgram.c:1498:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_recvmmsg:local address not available:crypto/bio/bss_dgram.c:1527:
        # OPENSSL_TEST_RAND_ORDER=1662544797
        not ok 4 - iteration 4
# ------------------------------------------------------------------------------
    # OPENSSL_TEST_RAND_ORDER=1662544797
    not ok 1 - test_bio_dgram
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/bio_dgram_test => 1
not ok 1 - running bio_dgram_test
# ------------------------------------------------------------------------------
#   Failed test 'running bio_dgram_test'
#   at /var/lib/buildbot/worker/ossl-freebsd13-x86_64/master_unix-freebsd13-x86_64/build/util/perl/OpenSSL/Test/Simple.pm line 77.
# Looks like you failed 1 test of 1.04-test_bio_dgram.t ................ 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
````

### PR DESCRIPTION
**Pull request by hlandau**
Fixes #19156.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
Close/open to have the buildbot workers kick in

**Comment by levitte** [comment]
Never mind buildbot/master:unix-centos8-x86_64 and bulidbot/master:unix-ubuntu2004-x86_64, they are currently paused.
I removed their jobs from the build request queues, but that doesn't seem to be reflected here, unfortunately.

**Comment by t8m** [comment]
OK with urgent.

File: crypto/bio/bss_dgram.c [hunk 1]
```diff
@@ -1518,16 +1531,14 @@ static int dgram_recvmmsg(BIO *b, BIO_MSG *msg,
...
-                }
-            }
+                              BIO_MSG_N(msg, stride, i).local) < 1)
+                /*
+                 * It appears BSDs do not support local addresses for
+                 * loopback sockets. In this case, just clear the local
+                 * address, as for OS X and Windows in some circumstances
+                 * (see below).
+                 */
+                BIO_ADDR_clear(msg->local);
```
Comment (levitte) [inline crypto/bio/bss_dgram.c:1541]: Isn't indentation a bit wonky here?

**Comment by hlandau** [comment]
Agree urgent.

**Comment by levitte** [comment]
Please go ahead

**Comment by hlandau** [comment]
Merging.

**Comment by hlandau** [comment]
Merged to master.

**Pull request closed without merging**